src/main/ast/Group.kt

```diff
@@ -0,0 +1,5 @@
+package ast
+
+abstract class Group(name: String) : Node(name){
```

Comment:
Is there any particular reason that a Group needs a name? If this is for typechecking we can use something like the [`is` keyword](https://kotlinlang.org/docs/reference/typecasts.html) to check types or even make a Group part of a [sealed class](https://kotlinlang.org/docs/reference/sealed-classes.html).

Reply:
Guess not; I thought it was part of the magic of typesafe builders...  Removed.